cryptographic_estimators/base_algorithm.py

```diff
@@ -338,7 +340,7 @@ def _fix_ranges_for_already_set_parameters(self):
         """
         parameters = self._optimal_parameters
         ranges = self._parameter_ranges
-        new_ranges = {i: ranges[i].copy() if i not in parameters else {"min": parameters[i], "max": parameters[i]}
+        new_ranges = {i: ranges[i].copy() if i not in parameters else {"min": parameters[i], "max": parameters[i] + 1}
```

Comment:
the original fix was not complete. We need to increase the upper bound by 1, s.t. `_valid_choices` iterates at least once.

Reply:
This change is not correct, the upper bound is generally handled as inclusive. If an SDAlgorithm overwrites the _valid_choices function it must take that into account. I see that many SD algorithms do not do this. This has to be changed instead. Otherwise algorithms of other estimators will consider to large ranges now.